key-rotator/main.go

```diff
@@ -336,8 +336,10 @@ func writeKeys(ctx context.Context, keyStore storage.Key, locality string,
 	// Write packet encryption key.
 	eg.Go(func() error {
 		if oldPacketEncryptionKey.Equal(newPacketEncryptionKey) {
+			log.Debug().Msgf("Skipping write for packet encryption key for %q: key unchanged", locality)
```

Comment:
I think instead of putting the locality into the log message, you could instead make it a field in the structured log message (I think with `log.Debug().String("locality", locality).Msg(...)` or thereabouts), and then it will be easier to search for messages.

Reply:
Same observation applies to the log statements on lines 342, 354, 357.

Reply:
I include the relevant information in both the structured and unstructured messages, since the default log view for GKE will not show the structured data at *all* -- all you get is the content of `Msgf` (though thankfully the GKE viewer does include a link to the normal GCP log viewer that does include structured data).